endoreg_db/services/pseudonym_service.py

```
@@ -0,0 +1,89 @@
+    try:
+        # Calculate the hash using the existing logic
+        patient_hash = logic.calculate_patient_hash(sensitive_meta, salt=logic.SECRET_SALT)
+        
+        # Persist the hash to the Patient model
+        with transaction.atomic():
+            patient.patient_hash = patient_hash
+            patient.save(update_fields=['patient_hash'])
+        
+        logger.info(f"Generated and persisted pseudonym for patient {patient.id}: {patient_hash[:8]}...")
+        
+        return patient_hash, True
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)